src/views/Timesheets/index.js

```diff
@@ -0,0 +1,539 @@
+import React, { useState } from 'react';
+import 'date-fns';
+import {
+  Grid,
+  Card,
+  CardContent,
+  CardHeader,
+  FormControl,
+  InputLabel,
+  Select,
+  Input,
+  MenuItem,
+  Button,
+  Typography,
+  TextField,
+  Snackbar,
+} from '@material-ui/core/';
+import MuiAlert from '@material-ui/lab/Alert';
+import DateFnsUtils from '@date-io/date-fns';
+import jobs from '../../services/jobs';
+import { MuiPickersUtilsProvider, TimePicker, DatePicker } from '@material-ui/pickers';
+import ShiftDisplay from './components/ShiftDisplay';
+import './timesheets.css';
+
+const Timesheets = (props) => {
+  const [userJobs, setUserJobs] = useState([]);
+  const [selectedDateIn, setSelectedDateIn] = useState(null);
+  const [selectedDateOut, setSelectedDateOut] = useState(null);
+  const [selectedJob, setSelectedJob] = useState(null);
+  const [shiftTooLong, setShiftTooLong] = useState(false);
+  const [shiftTooShort, setShiftTooShort] = useState(false);
+  const [timeOutIsBeforeTimeIn, setTimeOutIsBeforeTimeIn] = useState(false);
+  const [enteredFutureTime, setEnteredFutureTime] = useState(false);
+  const [hoursWorkedInDecimal, setHoursWorkedInDecimal] = useState(0.0);
+  const [userShiftNotes, setUserShiftNotes] = useState('');
+  const [isOverlappingShift, setIsOverlappingShift] = useState(false);
+  const [shiftDisplayComponent, setShiftDisplayComponent] = useState(null);
+  const [snackbarOpen, setSnackbarOpen] = useState(false);
+
+  const handleTimeOutIsBeforeTimeIn = (timeIn, timeOut) => {
```

Comment:
Could this (`handleTimeOutIsBefore...`) have a name that better matches its function- or could you split it into multiple functions?

Reply:
We could change it to something like handleTimeErrors